tests/__init__.py

```diff
@@ -353,40 +353,73 @@ def dbus_con(self):
     def dbus_con_sys(self):
         return self.dbus_test_case.dbus_con_sys
 
-    def start_impl_portal(self, params=None, portal=None):
-        """
-        Start the impl.portal for the given portal name. If missing,
-        the portal name is derived from the class name of the test, e.g.
-        ``TestFoo`` will start ``org.freedesktop.impl.portal.Foo``.
-        """
-        portal = portal or self.portal_name
-        self.p_mock, self.obj_portal = self.dbus_test_case.spawn_server_template(
-            template=f"tests/templates/{portal.lower()}.py",
-            parameters=params,
-            stdout=subprocess.PIPE,
-        )
-        flags = fcntl.fcntl(self.p_mock.stdout, fcntl.F_GETFL)
-        fcntl.fcntl(self.p_mock.stdout, fcntl.F_SETFL, flags | os.O_NONBLOCK)
-        self.mock_interface = dbus.Interface(self.obj_portal, dbusmock.MOCK_IFACE)
+    def _get_server_for_module(self, module, bustype):
+        try:
+            return self.busses[bustype][module.BUS_NAME]
+        except KeyError:
+            server = dbusmock.SpawnedMock.spawn_for_name(
+                module.BUS_NAME,
+                "/dbusmock",
+                dbusmock.OBJECT_MANAGER_IFACE,
+                bustype,
+                stdout=subprocess.PIPE,
+            )
```

Comment:
theoretically you could get that key error if you specify and invalid bus type though then everything else will fall over too I guess. a simple
```py
assert bustype in dbusmock.BusType
```
should do the trick here but I also accept a "meh".